### PR TITLE
revert #9125 / fix back-compatibility with saving hparams as a whole container

### LINKED ISSUE #9631: incorrect "hparam_name" saved to checkpoint when saving from a params in a container
## 🐛 Bug

When the hyperparameters get passed in as a single argument container and then saved using
`self.hyperparameters(container)`, the name of the argument does not get captured anymore as it was before. 

### To Reproduce
```python

import os

import torch
from argparse import Namespace
from torch.utils.data import DataLoader, Dataset

from pytorch_lightning import LightningModule, Trainer


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):
    def __init__(self, cfg: Namespace):
        super().__init__()
        self.save_hyperparameters(cfg)
        self.layer = torch.nn.Linear(32, 2)

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("train_loss", loss)
        return {"loss": loss}

    def configure_optimizers(self):
        return torch.optim.SGD(self.layer.parameters(), lr=0.1)


def run():
    train_data = DataLoader(RandomDataset(32, 64), batch_size=2)

    model = BoringModel(cfg=Namespace(a=2, b=3))
    trainer = Trainer(
        default_root_dir=os.getcwd(),
        limit_train_batches=1,
        limit_val_batches=1,
        num_sanity_val_steps=0,
        max_epochs=1,
        weights_summary=None,
    )
    trainer.fit(model, train_dataloader=train_data)

    checkpoint = torch.load(trainer.checkpoint_callback.best_model_path)
    print(checkpoint["hyper_parameters"])
    assert checkpoint["hparams_name"] == "cfg" 


if __name__ == "__main__":
    run()
```

### Expected behavior

Assertion should pass, but instead the value was "hparams". 

### Environment

git bisect shows that PR #9125 caused it. 
The commit was included in the 1.4.7 release and is blocking NVIDIA from upgrading their [NeMo code base](https://github.com/NVIDIA/NeMo) to the newest PL version.

### Additional context

reported on slack

### PR DESCRIPTION
**Pull request by awaelchli**
This reverts commit 904dde7573c97245b45e477631ece989bd8c01e9, added in #9125. 

Fixes #9631

The tests added in #9125 still pass. The new test added here fails on master and asserting the behavior how it was before #9125

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/utilities/parsing.py [hunk 1]
```diff
@@ -22,12 +22,8 @@
 from typing_extensions import Literal
 
 import pytorch_lightning as pl
-from pytorch_lightning.utilities import _OMEGACONF_AVAILABLE
```
Comment (awaelchli) [inline pytorch_lightning/utilities/parsing.py:25]: no changes to the file here except git revert the commit mentioned above.

**Review by ananthsub — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !
